src/pages/DataPlaneAuthReq.tsx

```diff
@@ -1,114 +1,125 @@
 
     useEffect(() => {
-        let error = null;
         if (!catalogPrefix || !originalUrl) {
-            error =
-                'Invalid URL parameters. Please contact support for assistance.';
+            setRedirectResult({
+                error: 'Invalid URL parameters. Please contact support for assistance.',
+            });
         } else if (gatewayAuthError) {
-            error = gatewayAuthError.toString();
-        } else if (gatewayUrl && gatewayAuthToken) {
-            // Validate that the hostname in the orig_url is a subdomain of the gateway_url.
-            // This is necessary in order to prevent malicious links using an `orig_url` parameter
-            // that sends a user's auth token to a 3rd party.
-            // Ideally we would also validate that the subdomain matches the subdomain of the task,
-            // but doing so will require that control-plane be aware of those subdomains, which
-            // won't be the case until we implement pet-names.
-            try {
-                const gatewayHost = new URL(gatewayUrl).hostname;
-                const origUrlHostname = new URL(originalUrl).hostname;
-                if (!origUrlHostname.endsWith(`.${gatewayHost}`)) {
-                    error = 'invalid `orig_url` parameter has invalid hostname';
-                }
-            } catch (e: unknown) {
-                console.error('invalid orig_url url parameter', e);
-                error =
-                    'invalid `orig_url` parameter cannot be parsed as a URL';
-            }
-
-            if (error === null) {
-                const newUrl = new URL('/auth-redirect', originalUrl);
-                newUrl.searchParams.append('token', gatewayAuthToken);
-                newUrl.searchParams.append('orig_url', originalUrl);
-
-                logRocketConsole('redirecting after successful auth', {
-                    newUrl,
-                    gatewayUrl,
-                    originalUrl,
-                });
-                window.location.replace(newUrl.toString());
-            }
-        }
+            setRedirectResult({ error: gatewayAuthError.toString() });
+        } else if (reactorUrl && reactorToken) {
+            getLiveSpecs_dataPlaneAuthReq(catalogPrefix).then(
+                (response) => {
+                    const hostnameLabelValue =
+                        response.data?.[0]?.shard_labels.find(
+                            (label) => label.name === SHARD_LABELS.HOSTNAME
+                        )?.value;
+
+                    if (response.error || !hostnameLabelValue) {
+                        setRedirectResult({
+                            error:
+                                response.error ??
+                                `Failed to validate 'orig_url' parameter.`,
+                        });
+
+                        return;
+                    }
+
+                    const gatewayHost = reactorUrl.hostname;
```

Comment:
Small tweak... can we set this below the return on line `71` since it cannot be used until after that check?